File: Claude/Intermediárias/longest_substring/longest_substring.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        char_index = {}
        max_length = 0
        start = 0

        for end, char in enumerate(s):
            if char in char_index and char_index[char] >= start:
                start = char_index[char] + 1
            else:
                max_length = max(max_length, end - start + 1)

            char_index[char] = end

        return max_length

# Exemplo de uso
solution = Solution()
print(solution.lengthOfLongestSubstring("abcabcbb"))  # Saída: 3
print(solution.lengthOfLongestSubstring("bbbbb"))     # Saída: 1
print(solution.lengthOfLongestSubstring("pwwkew"))    # Saída: 3

File: Claude/Intermediárias/longest_substring/longest_substring.cpp
#include <iostream>
#include <string>
#include <unordered_map>
#include <algorithm>

class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        std::unordered_map<char, int> char_index;
        int max_length = 0;
        int start = 0;

        for (int end = 0; end < s.length(); end++) {
            if (char_index.find(s[end]) != char_index.end() && char_index[s[end]] >= start) {
                start = char_index[s[end]] + 1;
            } else {
                max_length = std::max(max_length, end - start + 1);
            }

            char_index[s[end]] = end;
        }

        return max_length;
    }
};

// Exemplo de uso
int main() {
    Solution solution;
    std::cout << solution.lengthOfLongestSubstring("abcabcbb") << std::endl;  // Saída: 3
    std::cout << solution.lengthOfLongestSubstring("bbbbb") << std::endl;     // Saída: 1
    std::cout << solution.lengthOfLongestSubstring("pwwkew") << std::endl;    // Saída: 3
    return 0;
}
